overlord/snapstate/autorefresh.go

```diff
@@ -360,23 +381,30 @@ func (m *autoRefresh) Ensure() error {
 				return nil
 			}
 
-			// Check that we have reasonable delays between attempts.
-			// If the store is under stress we need to make sure we do not
-			// hammer it too often
-			if !m.lastRefreshAttempt.IsZero() && m.lastRefreshAttempt.Add(refreshRetryDelay).After(time.Now()) {
+			overrideDelay := false
+			err = m.launchAutoRefresh(overrideDelay)
+			if _, ok := err.(*httputil.PersistentNetworkError); ok {
+				return err
+			} else if errors.Is(err, tooSoonError{}) {
+				// ignore error, retry the auto-refresh later
 				return nil
 			}
 
-			err = m.launchAutoRefresh()
-			if _, ok := err.(*httputil.PersistentNetworkError); !ok {
-				m.nextRefresh = time.Time{}
-			} // else - refresh will be retried after refreshRetryDelay
+			m.nextRefresh = time.Time{}
```

Comment:
maybe this merits a comment:

// we launched the auto-refresh or hit an error that is not a persistent network error, reset nextRefresh